OpenRA.Mods.Common/ServerTraits/LobbySettingsNotification.cs

```diff
@@ -30,15 +30,15 @@ public void ClientJoined(OpenRA.Server.Server server, Connection conn)
 			var options = server.Map.Rules.Actors["player"].TraitInfos<ILobbyOptions>()
 				.Concat(server.Map.Rules.Actors["world"].TraitInfos<ILobbyOptions>())
 				.SelectMany(t => t.LobbyOptions(server.Map.Rules))
-				.ToDictionary(o => o.Id, o => o);
+				.Select(o => System.Tuple.Create(o.Id, o.Name)).Distinct().ToDictionary(o => o.Item1, o => o.Item2);
```

Comment:
It would be more readable, more efficient (enumerating the items only once instead of multiple times), and more intuitive (having later-defined items override earlier ones) to split this into two operations:

```csharp
var options = server.Map.Rules.Actors["player"].TraitInfos<ILobbyOptions>()
	.Concat(server.Map.Rules.Actors["world"].TraitInfos<ILobbyOptions>())
	.SelectMany(t => t.LobbyOptions(server.Map.Rules));

var optionNames = new Dictionary<string, string>();
foreach (var o in options)
	optionNames[o.Id] = o.Name;
```